rhoas-client/src/main/java/com/redhat/service/bridge/rhoas/RhoasClientImpl.java

```diff
@@ -0,0 +1,274 @@
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.openshift.cloud.api.kas.auth.models.AclBinding;
+import com.openshift.cloud.api.kas.auth.models.AclOperation;
+import com.openshift.cloud.api.kas.auth.models.AclPatternType;
+import com.openshift.cloud.api.kas.auth.models.AclPermissionType;
+import com.openshift.cloud.api.kas.auth.models.AclResourceType;
+import com.openshift.cloud.api.kas.auth.models.NewTopicInput;
+import com.openshift.cloud.api.kas.auth.models.Topic;
+import com.openshift.cloud.api.kas.auth.models.TopicSettings;
+import com.openshift.cloud.api.kas.models.ServiceAccount;
+import com.openshift.cloud.api.kas.models.ServiceAccountRequest;
+import com.redhat.service.bridge.rhoas.dto.TopicAndServiceAccountRequest;
+import com.redhat.service.bridge.rhoas.dto.TopicAndServiceAccountResponse;
+import com.redhat.service.bridge.rhoas.exceptions.RhoasClientException;
+
+import io.smallrye.mutiny.Uni;
+
+@ApplicationScoped
+public class RhoasClientImpl implements RhoasClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RhoasClientImpl.class);
+
+    @Inject
+    KafkasMgmtV1Client mgmtClient;
+
+    @Inject
+    KafkaInstanceAdminClient instanceClient;
+
+    @Override
+    public Uni<TopicAndServiceAccountResponse> createTopicAndConsumerServiceAccount(TopicAndServiceAccountRequest request) {
+        String topicName = request.getTopicName();
+        String serviceAccountName = request.getServiceAccountName();
+        LOG.info("Started creation of topic={} and serviceAccount={}...", topicName, serviceAccountName);
+        return Uni.createFrom().item(() -> Context.build(topicName, serviceAccountName))
+                .onItem().transformToUni(this::createTopic)
+                .onItem().transformToUni(this::createServiceAccount)
+                .onItem().transformToUni(this::createServiceAccountACLs)
+                .onItem().transform(RhoasClientImpl::toResponse);
+    }
+
+    private Uni<Context> createTopic(final Context ctx) {
+        return instanceClient.createTopic(new NewTopicInput().name(ctx.getTopicName()).settings(new TopicSettings().numPartitions(1)))
+                .onItem().invoke(t -> LOG.info("Created topic {}", t.getName()))
+                .onItem().transform(ctx::withTopic)
+                .onFailure().recoverWithUni(failure -> {
+                    String reason = "Error when creating topic " + ctx.getTopicName();
+                    LOG.error(reason, failure);
+                    return Uni.createFrom().item(ctx.withFailure(failure, reason));
+                });
+    }
+
+    private Uni<Context> revertTopicCreation(final Context ctx) {
+        LOG.info("Deleting topic {} to restore original state after failure", ctx.getTopicName());
+        return instanceClient.deleteTopic(ctx.getTopicName())
+                .onItem().transform(v -> {
+                    LOG.info("Deleted topic {}", ctx.getTopicName());
```

Comment:
@kostola Can you please ensure all placeholders in log statements are surrounded by `'`.

Reply:
Done :+1: